Forgot Password Functionality and Auth page fix.

SushantBhandari04 opened the pull request:
This PR adds a complete forgot-password workflow using a token-based reset mechanism. It includes:

A /forgot-password form where the user can enter their email.
A secure token generated via crypto.randomBytes() and stored in the DB.
A reset password email sent via Nodemailer with a token link.
A /reset-password page to input a new password.
Backend API to validate token, hash the password using bcrypt, and update it.
Basic client-side validations and error handling.

Fixes #22 

Also the tab change is also fixed. After signing up, the tab is changing to signin.
<img width="1892" height="987" alt="Screenshot 2025-08-03 143703" src="https://github.com/user-attachments/assets/1a1425ce-2901-4fd2-979a-14ae653c9b81" />
<img width="1900" height="969" alt="Screenshot 2025-08-03 143730" src="https://github.com/user-attachments/assets/a4086fe3-48db-402b-b8df-2df233b9a1ab" />
<img width="1497" height="388" alt="Screenshot 2025-08-03 143756" src="https://github.com/user-attachments/assets/2062b6b0-e608-4747-bc2d-f9574a0c2152" />
<img width="1894" height="981" alt="Screenshot 2025-08-03 143812" src="https://github.com/user-attachments/assets/a46f3824-d6b9-475d-b7d4-b9310a16df22" />

SushantBhandari04 commented:
@yashvikram30 Please review it

yashvikram30 commented:
hey @SushantBhandari04 just wanted to confirm one thing, the present backend is based on /api folder right? isn't the /backend folder useless?

SushantBhandari04 commented:
> hey @SushantBhandari04 just wanted to confirm one thing, i believe that the present backend is based on /api folder. do correct me if i'm wrong, but presently the /backend folder does not function and the apis there are just unused code, right?

Yes actually the /backend is of no use. The /api is being used for backend.

SushantBhandari04 commented:
> > > hey @SushantBhandari04 just wanted to confirm one thing, i believe that the present backend is based on /api folder. do correct me if i'm wrong, but presently the /backend folder does not function and the apis there are just unused code, right?
> > 
> > 
> > Yes actually the /backend is not being used. The /api is being used for backend.
> 
> would appreciate that you delete the unused folder/ code as well for clarity man. i'm grading this pr as the highest level, thanks for the cleaning up the mess

Sure I will do that

SushantBhandari04 commented:
> > > hey @SushantBhandari04 just wanted to confirm one thing, i believe that the present backend is based on /api folder. do correct me if i'm wrong, but presently the /backend folder does not function and the apis there are just unused code, right?
> > 
> > 
> > Yes actually the /backend is not being used. The /api is being used for backend.
> 
> would appreciate that you delete the unused folder/ code as well for clarity man. i'm grading this pr as the highest level, thanks for the cleaning up the mess

I have deleted the /backend folder. Also added a basic profie page as it was not there. Fixed the profile avatar in the navbar.

yashvikram30 commented:
thank you for the contribution, can you just confirm once that the entire code is functional?

SushantBhandari04 commented:
> thank you for the contribution, can you just confirm once that the entire code is functional?

Yes it is

yashvikram30 commented:
thanks for the contribution @SushantBhandari04

SushantBhandari04 commented:
> thanks for the contribution @SushantBhandari04

Thanks. Have you labelled this as gssoc and also the level
